pkg/newlog/cmd/newlogd.go

```diff
@@ -0,0 +1,1281 @@
+func formatAndGetMeta(appuuid string) string {
+	var appName string
+	if appuuid != "" {
+		for _, appD := range domainUUID { // cycle through the domainUUID map and find the UUID and appName
+			if appD.appUUID == appuuid {
+				appName = appD.appName
+				return appName
+			}
+		}
+	}
+	metaStr := logs.LogBundle{
+		DevID:      devMetaData.uuid,
+		Image:      devMetaData.curPart,
+		EveVersion: devMetaData.imageVer,
+	}
+	mapJmeta, _ := json.Marshal(metaStr)
+	return string(mapJmeta)
+}
+
+func getEveInfo() {
+	for {
+		devMetaData.curPart = agentlog.EveCurrentPartition()
+		if devMetaData.curPart == "Unknown" {
+			log.Errorln("currPart unknown")
+			time.Sleep(time.Second)
+			continue
+		} else {
+			break
+		}
+	}
+	for {
+		devMetaData.imageVer = agentlog.EveVersion()
+		if devMetaData.imageVer == "Unknown" {
+			log.Errorln("imageVer unknown")
+			time.Sleep(time.Second)
+			continue
+		} else {
+			break
+		}
+	}
+}
+
+func cleanGzipTempfiles(dir string) {
+	gfiles, err := ioutil.ReadDir(dir)
+	if err == nil {
+		for _, f := range gfiles {
+			if !f.IsDir() && strings.HasPrefix(f.Name(), tmpPrefix) && len(f.Name()) > len(tmpPrefix) {
+				err = os.Remove(dir + "/" + f.Name())
+				if err != nil {
+					log.Error(err)
+				}
+			}
+		}
+	}
+}
+
+// flush more often when we are going down by reading from watchdog log message itself
+func checkWatchdogRestart(entry *inputEntry) {
+	// source can be watchdog or watchdog.err
+	if strings.HasPrefix(entry.source, "watchdog") {
+		if strings.Contains(entry.content, "Retry timed-out at") {
+			entry.severity = "emerg"
+			syncToFileCnt = 1
+
+			// in case if the system does not go down, fire a timer to reset it to normal sync count
+			schedResetTimer = time.NewTimer(300 * time.Second)
+		}
+	}
+}
+
+func rankByInputCount(Frequencies map[string]uint64) pairList {
+	pl := make(pairList, len(Frequencies))
+	i := 0
+	for k, v := range Frequencies {
+		pl[i] = pair{k, v}
+		i++
+	}
+	sort.Sort(sort.Reverse(pl))
+	return pl
+}
+
+type pair struct {
+	Key   string
+	Value uint64
+}
+
+type pairList []pair
+
+func (p pairList) Len() int           { return len(p) }
+func (p pairList) Less(i, j int) bool { return p[i].Value < p[j].Value }
+func (p pairList) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+
+// generate top 10 contributor in total bytes from services
+func getDevTop10Inputs() {
+	if logmetrics.DevMetrics.NumBytesWrite == 0 {
+		return
+	}
+
+	top10 := make(map[string]uint32)
+	pl := rankByInputCount(devSourceBytes)
+	for i, p := range pl {
+		if i >= 10 {
+			break
+		}
+		top10[p.Key] = uint32(p.Value * 100 / logmetrics.DevMetrics.NumBytesWrite)
+	}
+	logmetrics.DevTop10InputBytesPCT = top10
+}
+
+func getPtypeTimestamp(timeStr string) timestamp.Timestamp {
+	t, err := time.Parse(time.RFC3339, timeStr)
+	if err != nil {
+		log.Fatal(err)
+	}
+	tt := timestamp.Timestamp{Seconds: t.Unix(), Nanos: int32(t.Nanosecond())}
+	return tt
```

Comment:
Can't you return a pointer here to avoid copying the lock?

Reply:
ok. let me try that to see if the yetus on this goes away